runtime/compiler/optimizer/EscapeAnalysis.hpp

```diff
@@ -458,6 +459,9 @@ class TR_EscapeAnalysis : public TR::Optimization
    bool     collectValueNumbersOfIndirectAccessesToObject(TR::Node *node, Candidate *candidate, TR::Node *indirectStore, TR::NodeChecklist& visited, int32_t baseChildVN = -1);
    void     checkDefsAndUses();
    bool     checkDefsAndUses(TR::Node *node, Candidate *candidate);
+   void     gatherUsesThroughAternary();
+   void     gatherUsesThroughAternary(TR::Node *node, TR::NodeChecklist& visited);
```

Comment:
If this is the recursive implementation body convention would say add something like Impl to its suffix and if it isn't meant to be part of the opt API we could consider hiding it as private / protected.

Reply:
> we could consider hiding it as private / protected

For some reason everything in `TR_EscapeAnalysis` is public. . . .  I'll see about hiding more of it.

Reply:
Andrew @andrewcraik, I've renamed the recursive implementation of `gatherUsesThroughAternary` to `gatherUsesThroughAternaryImpl` in https://github.com/eclipse/openj9/pull/5699/commits/d10e03a13121fc002836f50c90327408cdfd4f63

In https://github.com/eclipse/openj9/pull/5699/commits/48afe7397e38009460abb6dce0baf9cabb16d479, I made most of the method and fields `protected`, but needed to add two `friend` classes -- `TR_FlowSensitiveEscapeAnalysis` and `TR_LocalFlushElimination` -- that pick through the implementation of `TR_EscapeAnalysis`.  But I'm not really happy with it.  I'm inclined to do this more properly in a follow up issue.  Thoughts?

Reply:
> do this more properly

Just to clarify, I mean adjust the API so that `TR_FlowSensitiveEscapeAnalysis` and `TR_LocalFlushElimination` gather information from `TR_EscapeAnalysis` in a way that's a bit tidier.

Reply:
The extra encapsulation from protected is reasonable and fixing up the need for the friend class relationship can wait for the time being IMO.